user/user_test.go

```diff
@@ -40,31 +43,216 @@ func TestRESTClient_NewUser(t *testing.T) {
 		Comment:  "",
 	}
 
-	ctx := context.Background()
-
 	postParams := &products.PostUsersParams{
 		User:    uReq,
 		Context: ctx,
 	}
 
+	getParams := &products.GetUsersParams{
+		Context:  ctx,
+		Username: &exampleUser}
+
 	p.On("PostUsers",
 		postParams,
 		mock.AnythingOfType("runtime.ClientAuthInfoWriterFunc")).
 		Return(&products.PostUsersCreated{},
 			nil)
 
+	p.On("GetUsers", getParams, mock.AnythingOfType("runtime.ClientAuthInfoWriterFunc")).
+		Return(&products.GetUsersOK{
+			Payload: []*model.User{{Username: exampleUser}},
+		}, nil)
+
+	_, err := cl.NewUser(ctx, exampleUser, exampleEmail, "", examplePassword, "")
+
+	assert.NoError(t, err)
+
+	p.AssertExpectations(t)
+
+}
+
+func TestRESTClient_GetUser(t *testing.T) {
+	p := &mocks.MockClientService{}
+
+	c := &client.Harbor{
+		Products:  p,
+		Transport: nil,
+	}
+
+	cl := NewClient(c, authInfo)
+
+	ctx := context.Background()
+
+	t.Run("NoUsernameProvided", func(t *testing.T) {
+		emptyUserName := ""
+
+		_, err := cl.GetUser(ctx, emptyUserName)
+
+		assert.Error(t, err)
+
+		p.AssertExpectations(t)
+	})
+
+	t.Run("GetUser", func(t *testing.T) {
+		getParams := &products.GetUsersParams{
+			Context:  ctx,
+			Username: &exampleUser}
+
+		p.On("GetUsers", getParams, mock.AnythingOfType("runtime.ClientAuthInfoWriterFunc")).
+			Return(&products.GetUsersOK{
+				Payload: []*model.User{{Username: ""}},
+			}, nil)
+
+		_, err := cl.GetUser(ctx, exampleUser)
+
+		assert.Error(t, err)
+
+		p.AssertExpectations(t)
+	})
+
+	t.Run("UserNotFound", func(t *testing.T) {
```

Comment:
What does this test aim for? It's confusing to me that the test called "UserNotFound" returns a user called "nonexistent-user" 😄